app/models/incomeSourceDetails/IncomeSourceDetailsResponse.scala

```diff
@@ -38,11 +38,11 @@ case class IncomeSourceDetailsModel(mtdbsa:String,
     this.copy(property = property2, businesses = businesses2)
   }
 
-  val startingTaxYear: Option[Int] = (businesses.flatMap(_.firstAccountingPeriodEndDate) ++ property.flatMap(_.firstAccountingPeriodEndDate))
-    .map(_.getYear).sortWith(_ < _).headOption
+  val startingTaxYear: Int = (businesses.flatMap(_.firstAccountingPeriodEndDate) ++ property.flatMap(_.firstAccountingPeriodEndDate))
+    .map(_.getYear).sortWith(_ < _).headOption.getOrElse(throw new RuntimeException("User missing first accounting period information"))
```

Comment:
in the https://jira.tools.tax.service.gov.uk/browse/MISUV-2920 the account period should always be there, you do not need to throw an error and should be able to just use .head

Reply:
ah so the story was changing the handling of the behaviour, shouldn't we be handling the error and not just throwing it?

Reply:
In addition: when setting test data for audit and whatyouowe logic we pass None when passing PropertyDetailsTestConstants containing object. Throwing the exception allowed me to detect and correct code to properly pass the property object to complet tests.
As part of practicing defensive programming, assuming that data is always there is the source of all evil, hence the consensus to throw the exception. Hope that helps.

Reply:
appreciate the information jack however that doesnt answer my question, should we be handling this error rather than just throwing it?

Reply:
resolve this since message with James confirms were letting the system handle it